megatron/model/language_model.py

```diff
@@ -436,6 +436,9 @@ def state_dict_for_save_checkpoint(self, destination=None, prefix='',
                     destination, prefix, keep_vars)
         encoder_state_dict = self.encoder.state_dict_for_save_checkpoint(
                 destination, prefix, keep_vars)
+        if args.random_ltd:
+            from deepspeed.runtime.data_pipeline.data_routing.helper import remove_random_ltd_state_dict
+            encoder_state_dict = remove_random_ltd_state_dict(encoder_state_dict)
```

Comment:
I added following comment to clarify: When using random-LTD, it is required to call remove_random_ltd_state_dict during model checkpoint saving to transfer the random-LTD-wrapped layers back to original layers. This will help to remove the dependency to random-LTD inside the checkpoint, so that during evaluation or finetuning of the checkpoint there is no need to depend on random-LTD again.

Reply:
Resolved.